### ISSUE TITLE
Provide clear feedback when a hot reload has finished

### ISSUE THREAD

**Issue by InMatrix** [flutter/flutter#23316]
In the latest round of the "Build a Simple UI" study, we've seen two participants waiting for hot reload to finish when it, in fact, had finished already. It appears that the current feedback in the console is not clear enough (see below). This is a problem when the changes are subtle on the UI or the simulator is covered by another window. 

```
Initializing hot reload...
Syncing files to device iPhone 6s...
Reloaded 0 of 427 libraries in 690ms.
```

Proposal:

```
Initializing hot reload...
  Syncing files to device iPhone 6s...
  Reloaded 0 of 427 libraries in 690ms.
Hot reload completed. 
```
Cc: @gspencergoog @jayoung-lee @galeyang

**Comment by jonahwilliams** [comment]
Briefly reverted while I fix some integration tets

**Comment by jayoung-lee** [comment]
Just checking in -- was this feature added back?

**Comment by tvolkert** [comment]
@jonahwilliams was this done, then reverted?

**Comment by jonahwilliams** [comment]
Yeah, there are a few devicelab tests that are string matching the output. If someone were to adjust this, they'd also need to run through the hot reload tests and fix them in the same PR
